Foundation/testsuite/src/FileChannelTest.cpp

```diff
@@ -59,6 +59,33 @@ FileChannelTest::~FileChannelTest()
 }
 
 
+void FileChannelTest::testRotateNever()
+{
+	std::string name = filename();
+	try
+	{
+		AutoPtr<FileChannel> pChannel = new FileChannel(name);
+		pChannel->setProperty(FileChannel::PROP_ROTATION, "never");
+		pChannel->open();
+		Message msg("source", "This is a log file entry", Message::PRIO_INFORMATION);
+		for (int i = 0; i < 200; ++i)
+		{
+			pChannel->log(msg);
+		}
+		File f(name);
+		assertTrue (f.exists());
+		f = name + ".0";
+		assertTrue (!f.exists());
+	}
+	catch (...)
+	{
+		remove(name);
+		throw;
```

Comment:
You shall fail the test here instead of re-throwing exception.

Reply:
Thanks for the review! The rationale for copying the pattern from the other similar tests was that it would primarily test that the log isn't rotated (or at least a simple case of that), and as a bonus it also fails on the bug that's been fixed. Changing the throw would deviate from the pattern, and make it less like the other tests.
Shall I go through with the change regardless?

Reply:
I see. I did not look at other test cases in the same file. I agree that it is better to stick to the pattern.